### PR TITLE
Updated Metric formatting to be more efficient and handle very small doubles

### PR DESCRIPTION
**Pull request by tfenne**
A fix for #318.

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/util/Metric.scala [hunk 1]
```diff
@@ -225,11 +228,25 @@ trait Metric extends Product with Iterable[(String,String)] {
...
 
+  /** @deprecated use [[formatValue]] instead. */
+  @deprecated("Use formatValue instead.", since="0.5.0")
+  protected def formatValues(value: Any): String = formatValue(value)
+
   /** Override this method to customize how values are formatted. */
-  protected def formatValues(value: Any): String = formatter.format(value)
+  protected def formatValue(value: Any): String = value match {
+    case null      => ""
+    case None      => ""
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/util/Metric.scala]: `ReflectionUtil.SpecialEmptyOrNoneToken`?

File: src/main/scala/com/fulcrumgenomics/util/Metric.scala [hunk 2]
```diff
@@ -225,11 +228,25 @@ trait Metric extends Product with Iterable[(String,String)] {
...
+    case Some(x)   => formatValue(x)
+    case d: Iso8601Date => d.toString
+    case d: Date   => Metric.DateFormat.synchronized { Metric.DateFormat.format(d) }
+    case f: Float  => formatValue(f.toDouble)
+    case d: Double if d < 0.00001 => Metric.SmallDoubleFormat.synchronized { Metric.SmallDoubleFormat.format(d) }
+    case d: Double                => Metric.BigDoubleFormat.synchronized { Metric.BigDoubleFormat.format(d) }
+    case other     => other.toString
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/util/Metric.scala]: align the `=>`?

File: src/main/scala/com/fulcrumgenomics/util/Metric.scala (hunk 1 above)
Comment (tfenne) [reply src/main/scala/com/fulcrumgenomics/util/Metric.scala]: I don't think we want that when _formatting_.  That'd result in `:none:` getting written into the text files and I think that'd be confusing.

Comment (nh13) [reply src/main/scala/com/fulcrumgenomics/util/Metric.scala]: Sorry, right, this is `Metric`.

**Pull request merged**